connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -7227,6 +7227,112 @@ object functions {
    */
   def to_csv(e: Column): Column = to_csv(e, Collections.emptyMap())
 
+  // scalastyle:off line.size.limit
+  /**
+   * Parses a column containing a XML string into the data type corresponding to the specified
+   * schema. Returns `null`, in the case of an unparseable string.
+   *
+   * @param e
+   *   a string column containing XML data.
+   * @param schema
+   *   the schema to use when parsing the XML string
+   * @param options
+   *   options to control how the XML is parsed. accepts the same options and the XML data source.
+   *   See <a href=
+   *   "https://spark.apache.org/docs/latest/sql-data-sources-xml.html#data-source-option"> Data
+   *   Source Option</a> in the version you use.
+   * @group collection_funcs
+   *
+   * @since 4.0.0
+   */
+  // scalastyle:on line.size.limit
+  def from_xml(e: Column, schema: StructType, options: Map[String, String]): Column =
```

Comment:
the schema parameter can be `StructType` or `Column`, the options parameter can be scala or java map, or omitted. This means we need 6 overloads of `from_xml`.

Does it really worth it? I know we did the same thing for `from_json`, but this is really convoluted.

How about something like
```
TextParsingFunction.newBuilder()
  .withSchema(...) // It has multiple overloads
  .withOptions(...) // It has multiple overloads
  .xml() // returns a Column
```

Anyway, it's unrelated to this PR. We can do it later. cc @HyukjinKwon

Reply:
I think we can remove `(Scala-specific)` signature for now. For the same reason, we don't have that scala specific version of `to_csv`, etc.

Reply:
from_csv has just two. I can trim from_xml overloads too. Let me know.

Reply:
Let's remove this signature with Scala map for now in a followup.